platform/osRegistryInterface.go

```diff
@@ -0,0 +1,46 @@
+//go:build windows
+// +build windows
+
+package platform
+
+import "golang.org/x/sys/windows/registry"
+
+// Registry interface for interacting with the Windows registry
+type Registry interface {
+	OpenKey(k registry.Key, path string, access uint32) (RegistryKey, error)
+}
+
+// RegistryKey interface to represent an open registry key
+type RegistryKey interface {
+	GetStringValue(name string) (string, uint32, error)
+	SetStringValue(name, value string) error
+	Close() error
+}
+
+type WindowsRegistry struct{}
+
+// WindowsRegistryKey implements the RegistryKey interface
+type WindowsRegistryKey struct {
+	key registry.Key
+}
+
+func (r *WindowsRegistry) OpenKey(k registry.Key, path string, access uint32) (RegistryKey, error) {
```

Comment:
`k` is unused here--I don't think that's intentional. Granted, we're probably always going to use HKLM, but who knows? If we're making a generic wrapper, I'd like to be able to, say, edit some keys in HKCU.

Reply:
as someone with no understanding of win reg, you could be making these up and I would have no idea 😂

Reply:
@rbtr this is cursed knowledge from working summers at my high school playing student sysadmin :-P The Norton Ghosts still haunt me.